### PR TITLE
Move API definitions to separate package

### PR DESCRIPTION
**Pull request by neoaggelos**
### Summary

Requires https://github.com/canonical/k8s-snap-api/pull/2

### Changes

- Remove `api/v1` from the k8s-snap package
- Add `github.com/canonical/k8s-snap-api` dependency
- Use the new `apiv1` package everywhere
- Keep the `k8s status` formatting code in `k8s_status_format.go` and tests (these are not part of the API)
- Move `microcluster` <-> BootstrapConfig and JoinConfig converters into the `utils` package (the names we use to pass these configs in the microcluster map are not part of the API.
- Adjust `k8sd.Client` interface to consistently work with request and response messages.

### Notes

This is currently using a test tag from https://github.com/canonical/k8s-snap-api. We first need to merge https://github.com/canonical/k8s-snap-api/pull/2 and then cut a `v0.1.0` tag

### CODE REVIEW

**Review by eaudetcobello — APPROVED** [review]

File: src/k8s/cmd/k8s/k8s_x_snapd_config.go [hunk 1]
```diff
@@ -58,7 +58,7 @@ func newXSnapdConfigCmd(env cmdutil.ExecutionEnvironment) *cobra.Command {
 					env.Exit(1)
 					return
 				}
-				if err := snapdconfig.SetSnapdFromK8sd(cmd.Context(), config, env.Snap); err != nil {
+				if err := snapdconfig.SetSnapdFromK8sd(cmd.Context(), config.Config, env.Snap); err != nil {
```
Comment (berkayoz) [inline src/k8s/cmd/k8s/k8s_x_snapd_config.go:61]: Should this also be in the form of `response.Config` way?

File: src/k8s/cmd/k8s/k8s_kubectl.go [hunk 2]
```diff
@@ -24,15 +24,15 @@ func newKubectlCmd(env cmdutil.ExecutionEnvironment) *cobra.Command {
 				return
 			}
 
-			if status, initialized, err := client.NodeStatus(cmd.Context()); err != nil {
+			if resp, initialized, err := client.NodeStatus(cmd.Context()); err != nil {
```
Comment (berkayoz) [inline src/k8s/cmd/k8s/k8s_kubectl.go:27]: I see `resp` and `response` mixed in the `cmd`, should we unify those to be one of the 2?

File: src/k8s/pkg/k8sd/api/kubernetes_auth_tokens.go [hunk 3]
```diff
@@ -8,30 +8,14 @@ import (
...
-		var err error
-		username, groups, err = database.CheckToken(ctx, tx, token)
-		return err
-	}); err != nil {
-		return response.NotFound(err)
-	}
-
-	return response.SyncResponse(true, apiv1.CheckKubernetesAuthTokenResponse{Username: username, Groups: groups})
-}
-
```
Comment (berkayoz) [inline src/k8s/pkg/k8sd/api/kubernetes_auth_tokens.go:34]: I assume we are dropping token auth hook here, just making sure.

**Review by berkayoz — COMMENTED** [review]
Great work, overall looks good to me. Left some comments.

File: src/k8s/pkg/k8sd/api/kubernetes_auth_tokens.go (hunk 3 above)
Comment (neoaggelos) [reply src/k8s/pkg/k8sd/api/kubernetes_auth_tokens.go:34]: this is just a testing endpoint, it is not used anywhere. the token review webhook is still there and works

**Review by berkayoz — APPROVED** [review]

**Pull request merged**